### PR TITLE
Test/add unittests

### PR DESCRIPTION
**Pull request by csc-jm**
### Description

Add unit tests 

### Changes Made

- Basic unit tests implemented 
- xml/xsd files for testing included
- tox automation included for running tests and flake8 checks 

### Testing

- [x] Unit Tests

### CODE REVIEW

**Comment by csc-jm** [comment]
Had to include some files for testing, as I realized the initial version of the CLI is not able to gather xml files from a web url yet. Is this a functionality that the tool could use however?

File: tests/test_validator.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+    def tearDown(self):
+        """Remove setup variables."""
+        pass
+
+    def test_no_args(self):
+        """Test case where no args are passed."""
+        result = self.runner.invoke(cli, [])
+        assert result.exception  # Catch any exception
+        assert result.exit_code == 2  # Catch specifically a usage error
```
Comment (blankdots) [inline tests/test_validator.py:31]:
```suggestion
        self.assertEqual(result.exit_code, 2)  # Catch specifically a usage error
```

if you are using the unit test class, do this for other `asserts` as well.

File: tests/test_validator.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        self.xsd_path = self.TESTFILES_ROOT / 'schemas'
+
+    def tearDown(self):
+        """Remove setup variables."""
+        pass
+
+    def test_no_args(self):
+        """Test case where no args are passed."""
+        result = self.runner.invoke(cli, [])
+        assert result.exception  # Catch any exception
```
Comment (blankdots) [inline tests/test_validator.py:30]: not sure this is needed to be checked.

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+max-line-length = 80
+
+[testenv:flake8]
+deps =
+    flake8
+commands = flake8 .
+
+[testenv]
+deps =
+    -r{toxinidir}/requirements.txt
```
Comment (blankdots) [inline tox.ini:22]:
```suggestion
    -rrequirements.txt
```

File: tests/test_validator.py (hunk 2 above)
Comment (csc-jm) [reply tests/test_validator.py:30]: checking if any exception is raised or checking the no args case?

Comment (blankdots) [reply tests/test_validator.py:30]: then that is not the proper way you need: https://docs.python.org/3/library/unittest.html?highlight=assertcountequal#unittest.TestCase.assertRaises

Comment (blankdots) [reply tests/test_validator.py:30]: it is how they recommend here: https://click.palletsprojects.com/en/7.x/testing/ but you are using a unittest class, so use what is provided by that (as I mention: https://docs.python.org/3/library/unittest.html?highlight=assertcountequal#classes-and-functions)

**Comment by blankdots** [comment]
related to:
> Had to include some files for testing, as I realized the initial version of the CLI is not able to gather xml files from a web url yet. Is this a functionality that the tool could use however?

in this initial version no, probably start an issue on it and add it as a feature.

File: tests/test_validator.py (hunk 2 above)
Comment (csc-jm) [reply tests/test_validator.py]: Yea, so I originally noticed that the assertRaises() method prints the CLI output as well, which made running the tests a bit messy. The latest commit now just checks the exit codes from each test which I imagine is conclusive enough?

**Pull request merged**